helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -221,14 +221,19 @@ public void finalize() {
     close();
   }
 
-  public void addCluster(String clusterName, boolean overwritePrevious, CloudConfig cloudConfig)
+  public void addCluster(String clusterName, boolean overwritePrevious, CloudConfig cloudConfig) {
+    addCluster(clusterName, Arrays.asList(BuiltInStateModelDefinitions.values()), overwritePrevious, cloudConfig);
+  }
+
+  public void addCluster(
+      String clusterName, List<BuiltInStateModelDefinitions> stateModelDefs, boolean overwritePrevious, CloudConfig cloudConfig)
```

Comment:
Why we need this?

Reply:
Yes, so we are creating a function here that creates the cluster loading by selectively loading the state-model defs. 

This is needed for the an optimization that we are utilizing during every test cluster creation. We have found that loading all state-model defs everytime takes about 500-700ms, due to invocation of ZKUtil::isClusterSetup function
that (repeatedly) ensure all the helix cluster paths are created before loading a state-model.

I have added as part of the java doc as well.